docs/Users/Tutorials/Linter_Bears_Advanced.rst

```diff
@@ -0,0 +1,134 @@
+combined for different output streams. Specifying an own output
+parsing/processing behaviour, ``process_output`` can be overridden.
+
+::
+
+    @linter(executable='my_tool')
+    class MyBear:
+        def process_output(self, output, filename, file):
+            pass
+
+The ``output`` variable contains the string output from the executable.
+Depending on how you use the ``use_stdout`` and ``use_stderr`` parameters from
+``@linter``, ``output`` can contain either a tuple or a plain string: If
+``use_stdout`` and ``use_stderr`` are both ``True``, a tuple is placed with
+``(stdout, stderr)``. If only one of them is ``True``, a string is passed
+(containing the regarding activated output stream).
+
+Inside ``process_output`` you need to yield results according to the executable
+output. It is also possible to combine the built-in capabilities. There are
+several functions accessible with the naming scheme
+``process_output_<output-format>``.
+
+- ``process_output_regex``: Extracts results using a regex.
+- ``process_output_corrected``: Extracts results (with patches) by using a
+  corrected version of the file processed.
+
+::
+
+    @linter(executable='my_tool',
+            use_stdout=True,
+            use_stderr=True)
+    class MyBear:
+        # Assuming the tool puts a corrected version of the file into stdout
+        # and additional issue messages (that can't be fixed automatically)
+        # into stderr, let's combine both streams!
+        def process_output(self, output, filename, file):
+            # output is now a tuple, as we activated both, stdout and stderr.
+            stdout, stderr = output
+            yield from self.process_output_corrected(stdout, filename, file)
+            regex = "(?P<message>.*)"
+            yield from self.process_output_regex(stderr, filename, file, regex)
+
+JSON output is also very common:
+
+::
+
+    @linter(executable='my_tool')
+    class MyBear:
+    def process_output(self, output, filename, file):
+        for issue in json.loads(output):
+            yield Result.from_values(origin=self,
+                                     message=issue["message"],
+                                     file=filename)
+
+Additional Prerequisite Check
+-----------------------------
+
+``@linter`` supports doing an additional executable check before running the
+bear, together with the normal one (checking if the executable exists). For
+example, this is useful to test the existence of external modules (like Java
+modules).
+
+To enable this additional check with your commands, use the
+``prerequisite_check_command`` parameter of ``@linter``.
+
+::
+
+    @linter(executable='...'
+            prerequisite_check_command=('python3', '-c', 'import my_module'))
+    class MyBear:
+        pass
+
+If the default error message does not suit you, you can also supply
+``prerequisite_check_fail_message`` together with
+``prerequisite_check_command``.
```

Comment:
hm not sure, imo it does not make sense after this sentence to supply one without `check_command`. And if someone tries he will get an error anyway :3